mlcomp/parallelm/ml_engine/sagemaker_engine.py

```diff
@@ -47,29 +50,63 @@ def iam_role(self):
     def _setup_iam_role(self, eng_args_config):
         self._iam_role = EeArg(eng_args_config.get('iam_role')).value
         if not self._iam_role:
-            # acount_id = boto3.client('sts').get_caller_identity().get('Account')
-
-            now = datetime.datetime.now()
-            role_name = 'AmazonSageMaker-ExecutionRole-{y}{month}{d}T{h}{minute}{s}' \
-                .format(y=now.year, month=now.month, d=now.day, h=now.hour, minute=now.minute, s=now.second)
-
-            tags = [{'Key': self._tag_key, 'Value': self._tag_value}] if self._tag_key else []
-
-            client = boto3.client('iam')
-
-            try:
-                response = client.create_role(
-                    Path=SageMakerEngine.SERVICE_ROLE_PATH,
-                    RoleName=role_name,
-                    AssumeRolePolicyDocument='<URL-encoded-JSON>',
-                    Description='Auto generated sagemaker aim role by ParallelM',
-                    Tags=tags
-                )
-                self._iam_role_name = response[SageMakerEngine.ROLE_RESPONSE_GROUP][SageMakerEngine.ROLE_RESPONSE_NAME_KEY]
-                self._iam_role = response[SageMakerEngine.ROLE_RESPONSE_GROUP][SageMakerEngine.ROLE_RESPONSE_ARN_KEY]
-            except ClientError as e:
-                self._logger.error("Failed to create a an iam role for sagemaker service!\n{}".format(e))
-                raise e
+            self._create_role()
+
+    def _create_role(self):
+        """
+        Creates an IAM role for SageMaker service (
+        https://docs.aws.amazon.com/sagemaker/latest/dg/sagemaker-roles.html#sagemaker-roles-amazonsagemakerfullaccess-policy)
+
+        Please note that in order to be able to execute this function, proper permissions should be set upfront for
+        the given user. To set permissions, do the following:
+        1. Enter to AWS console ==> "My Security Credentials" ==> Users ==> <relevant-user> ==> "Add inline policy"
+        2. set the following write permissions:
+            "iam:DetachRolePolicy", "iam:CreateRole", "iam:DeleteRole", "iam:AttachRolePolicy"
+        3. Provides a name to the new attached permissions and save
+        4. Make sure there are no any other contradict permission policy that block one of the permissions above.
+        """
+        now = datetime.datetime.now()
+        role_name = 'AmazonSageMaker-ExecutionRole-{y}{month}{d}T{h}{minute}{s}' \
+            .format(y=now.year, month=now.month, d=now.day, h=now.hour, minute=now.minute, s=now.second)
+
+        tags = [{'Key': self._tag_key, 'Value': self._tag_value}] if self._tag_key else []
+
+        trust_policy = {
+            "Version": "2012-10-17",
```

Comment:
constant?

Reply:
Not really. This is a signature that comes from AWS SageMaker and expected to be seen in the given json as is.